### ISSUE TITLE
[Request] Move away from themoviedb.org

### ISSUE THREAD

**Issue by Xperplay** [elgatito/plugin.video.elementum#245]
Hi,

First off I do realize Pulsar/Quasar/Elementum are built on themoviedb.org so this might be a challenging task but themovedb.org is very unreliable.

I have just quickly compared my Trakt TV Shows collection to my kodi library managed by Elementum. I did not go into every season to check for completion. Here are my results...

Show Title: **Strip the Cosmos**
Issue: Completely Missing from Library
TMDb URL: https://www.themoviedb.org/tv/68813-strip-the-cosmos?language=en-US
Trakt URL: https://trakt.tv/shows/strip-the-cosmos

Show Title: **Horizon**
Issue: Missing 5 episodes missing from latest season
TMDb URL: https://www.themoviedb.org/tv/1966-horizon/seasons?language=en-US
Trakt URL: https://trakt.tv/shows/horizon/seasons/2018

Show Title: **Extreme Universe**
Issue: Completely Missing from Library
TMDb URL: https://www.themoviedb.org/tv/38673-extreme-universe?language=en-US
Trakt URL: https://trakt.tv/shows/extreme-universe

Show Title: **Nova**
Issue: Missing latest episode
TMDb URL: https://www.themoviedb.org/tv/3562-nova/season/45?language=en-US
Trakt URL: https://trakt.tv/shows/nova/seasons/45

Show Title: **Super Game Jam**
Issue: Completely Missing from Library
TMDb URL: N/A
Trakt URL: https://trakt.tv/shows/super-game-jam

Show Title: **Origins**
Issue: Completely Missing from Library
TMDb URL: N/A
Trakt URL: https://trakt.tv/shows/origins

Keep in mind this is a very small sample so the amount of missing content on a larger scale would be huge. If you know of other missing content from TMDb please post.

Nova and Horizon are seasons which are currently airing and episodes are missing from months ago. TMDb is user maintained which makes it very unreliable. I ask you what is the point of having all this automation when at the end it all depends on some random person on the internet inputting data on TMDb?

I am not sure why TMDb show numbers are used in .strm files, it seems like an unnecessary step. Why not just use the show title?

P.S: Other addons do not have this issue, I ended up using Neptune Rising to manage my TV Shows library and then use context helper to play with Elementum which is not ideal at all.

**Comment by elgatito** [comment]
I know there are better sources for anime, maybe for some other content types, but we use TMDB as a main information source and unified identifying location.

Regarding Trakt - it's not free, it can become payable at any moment, and the main issue - it does not have translations. It's fine if you use English, but most of users don't.

This addon will stay of TMDB. I don't see benefits worth spending huge ammount of time for that.

**Comment by Xperplay** [comment]
tvmaze.com is a better alternative and I believe it is free. It also uses similar url format to TMDb so maybe it does not require that much work. Missing episodes from currently airing shows makes the whole library integration unusable imo.

**Comment by sjimmie** [comment]
Hi,

If you go to
https://www.themoviedb.org/talk/category/504794a319c29527aa000640
You are able to log cases for every show / episode you believe the content
is wrong or missing.
After they updated their DB your problem should be fixed.

**Comment by elgatito** [comment]
> Yes but once again I have to ask what is the point of having all this automation when at the end it all depends on some random person on the internet inputting data on TMDb? This is a small sample there is tons of missing content on TMDb

You understand you want to move from crowded community service (TMDB) to a less-crowded community service (tvmaze)?

**Comment by sjimmie** [comment]
Yes but once again I have to ask what is the point of having all this
automation when at the end it all depends on some random person on the
internet inputting data on TMDb?   -> That is what open-source is all about.

If you don't want that, find a paid service that does what you want and pay
for it :-)

**Comment by Xperplay** [comment]
@elgatito Maybe less crowded but more complete and that is all that matters at the end of the day.

Another possible workaround:

Can Elementum handle titles in strm files at the moment or would it be easy to implement? I ask because I can search for torrents using a title trough Elementum so it seems the feature is there. Also when I use the context helper I beleive the title is passed to Elementum search. Can I make a strm file that uses Elementum search feature? 

Because that would be a simple work around. Could add an option for Elementum to create strm files with titles.

**Comment by elgatito** [comment]
> Can Elementum handle titles in strm files at the moment or would it be easy to implement? 

You don't understand how it works. Title is used later for Kodi scrapers to find proper item. 
Strm file contains TMDB item id, to know which item will be played, make query for it and do search within providers. It is needed to perform proper query generation and then use it.

**Comment by Xperplay** [comment]
Then how does the context helper work without a TMDb id? It seems a query can be made without a TMDb id

**Comment by elgatito** [comment]
> Then how does the context helper and Elementum search feature work without a TMDb id? 

It's trying to find TMDB id to use it later.


> It seems a query can be made without a TMDb id

It's a plain search query in that case. In short - you can't use queries in strm files. Use other addons.

**Comment by Xperplay** [comment]
I noticed thetvdb is used in the nfo files. It is also more complete than TMDb.

I tried making strm files with:
`plugin://plugin.video.elementum/search?q=query`

This is very close to working properly, Elementum performs the search , results displayed, after buffering ends the video starts playing but an error is displayed immediately  "One or more items failed to play" even tough the video is playing fine. When clicking OK on the dialog box the video is stopped if it is not clicked the video keeps playing behind the dialog box.

I also tried making strm files with:
` http://127.0.0.1:65220/search?q=query`

This is also close to working properly, the only issue here is that Elementum search is triggered twice. Results dialog is displayed twice. Which works fine if I cancel the second results dialog.


Lots of options here, surely one of these can be implemented/fixed
